components/vf-masthead/vf-masthead--inlay.hbs

```diff
@@ -1,25 +1,29 @@
-<div class="vf-inlay">
-  <header class="vf-inlay__header">
-    <div class="vf-masthead">
-      <div class="vf-masthead__inner">
-        <div class="vf-masthead__title">
-          <h1 class="vf-masthead__heading">
-            <a class="vf-masthead__heading__link" href="http://dev-vf-theme-prototype.pantheonsite.io">Häring Group</a>
-          </h1>
-        </div>
-      </div>
+<style>
+:root {
+  /* --vf-masthead__color--background: none; */
+  /* --vf-masthead__color--foreground: #000000; */
+  /* --vf-masthead__bg-image: url('http://www.fillmurray.com/800/200'); */
+}
+</style>
+
+<header class="vf-inlay__header">
+  <div class="vf-masthead" style="background-color: var(--vf-masthead__color--background, #000000); color: var(--vf-masthead__color--foreground, #ffffff);">
```

Comment:
I'm guessing defaults for these (`--vf-masthead__color--background`) would be set in the Sass somewhere, ya?

Reply:
So the custom properties in the `:root` Would need to be added to the document after the css is linked in the head and before the component. 

The defaults are inline in the mark up (black and white in the example), so if there’s no theme or that bit doesn’t get included there’s a ‘nice’ default (of course black and white is just here as an example. It could easily be green white.)

This pattern also allows for background images too. So I think it’s a more general ‘masthead’ pattern with its parent dictating how wide it is. For example. In vf-inlay it would only need some padding on the parent header really.

Reply:
Shouldn't we also set some sort of global default in the VF so that way if someone forgets to add a local `style=""` there's a reasonable background -- that and it would help if the EMBL green changes so we won't have the default colour inlined in all the sites.

Reply:
Well, we can set another custom variable in the stylesheet. That would be the back up colour in the `var` rather than being explicit with black/white.

```
<div 
  class=“vf-masthead” 
  style=“
  background-color: var(—custom-defined-bg-color, var(—backup-bg-color))
  “
>
```

(The syntax might be off there - but you should get the idea)

Reply:
I think maybe we could do what I'm talking about if we had in the pattern sass:

```sass
:root {
--vf-masthead__color--background: #{set-color(vf-color-green)};
}

...

.vf-masthead {
  background-color: var(--vf-masthead__color--background);

```

And then did in the pattern html
```html
<style>
:root {
  /* --vf-masthead__color--background: none; */
  ...
</style> 
  <div class="vf-masthead" style="background-color: var(--vf-masthead__color--background); ...
``` 

That should allow the VF default to be used, unless the variable has been overridden prior to the html.

Reply:
Comment overlap 🤸‍♂️

Reply:
so, the only 'nice' way to do this (because I cannot see a way to put a Sass variable from a Sass map using a Sass function that works). Is by duplicating the Sass variables for colour as custom properties.

I think moving forward that'd be a good thing anyways.